### PR TITLE
[oxygen] Merge forward from 2017.7 to oxygen

### PR DESCRIPTION
**Pull request by rallytime**
Conflicts:
  - salt/fileclient.py
  - salt/grains/disks.py
  - salt/modules/cmdmod.py
  - salt/state.py
  - salt/states/file.py
  - salt/utils/gitfs.py
  - salt/utils/win_functions.py
  - tests/integration/modules/test_state.py
  - tests/unit/cache/test_localfs.py

### CODE REVIEW

File: salt/modules/cmdmod.py [hunk 1]
```diff
@@ -52,8 +52,10 @@
 
 if salt.utils.platform.is_windows():
     from salt.utils.win_runas import runas as win_runas
+    from salt.utils.win_functions import escape_argument as _cmd_quote
     HAS_WIN_RUNAS = True
 else:
+    from salt.ext.six.moves import shlex_quote as _cmd_quote
```
Comment (terminalmage) [inline salt/modules/cmdmod.py]: The `oxygen` branch added the cmd_quote import on line 45 above, but this is a better fix for this, so please remove line 45.

File: salt/utils/gitfs.py [hunk 2]
```diff
@@ -2829,17 +2834,24 @@ def file_hash(self, load, fnd):
...
+            if exc.errno != errno.ENOENT:
+                raise exc
+
+        try:
+            os.makedirs(os.path.dirname(hashdest))
+        except OSError as exc:
+            if exc.errno != errno.EEXIST:
+                raise exc
+
+        ret['hsum'] = salt.utils.get_hash(path, self.opts['hash_type'])
```
Comment (terminalmage) [inline salt/utils/gitfs.py]: This should be `salt.utils.hashutils.get_hash`

**Review by terminalmage — CHANGES_REQUESTED** [review]

File: salt/modules/cmdmod.py (hunk 1 above)
Comment (rallytime) [reply salt/modules/cmdmod.py]: 👍

File: salt/utils/gitfs.py (hunk 2 above)
Comment (rallytime) [reply salt/utils/gitfs.py]: Ah, I forgot to run the check for the moved utils. Thanks!

**Review by terminalmage — APPROVED** [review]

**Pull request merged**